packages/flutter/lib/src/animation/animation_controller.dart

```diff
@@ -565,7 +568,8 @@ class AnimationController extends Animation<double>
       }
       return true;
     }());
-    return animateWith(_RepeatingSimulation(min, max, period));
+    assert(reverse != null);
```

Comment:
also need to assert that max  <= upperBound and min >= lowerBound.

Reply:
Also done 😉